File: 프로그래머스/0/181898. 가까운 1 찾기/가까운 1 찾기.cpp
#include <vector>

using namespace std;

int solution(vector<int> arr, int idx) {
    // idx부터 시작하여 배열 끝까지 순회
    for (int i = idx; i < arr.size(); i++) {
        if (arr[i] == 1) {
            return i; // 가장 먼저 발견된 1의 인덱스 즉시 반환
        }
    }
    
    // 1을 찾지 못한 경우 -1 반환
    return -1;
}

File: 프로그래머스/0/181898. 가까운 1 찾기/가까운 1 찾기.py
def solution(arr, idx):
    # idx부터 arr의 끝까지 탐색
    for i in range(idx, len(arr)):
        if arr[i] == 1:
            return i  # 가장 먼저 발견된 1의 인덱스 즉시 반환
            
    # 조건에 맞는 인덱스가 없으면 -1 반환
    return -1